tasks/eva_2779/correct_assembly_error_in_rat.py

```diff
@@ -0,0 +1,268 @@
+            result_insert = collection.with_options(write_concern=WriteConcern(w="majority", wtimeout=1200000))\
+                                      .bulk_write(requests=insert_statements, ordered=False)
+            total_inserted += result_insert.inserted_count
+            result_drop = collection.with_options(write_concern=WriteConcern(w="majority", wtimeout=1200000)) \
+                                    .bulk_write(requests=drop_statements, ordered=False)
+            total_dropped += result_drop.deleted_count
+        logger.info(f'{total_inserted} new documents inserted in {collection_name}')
+        logger.info(f'{total_dropped} old documents dropped in {collection_name}')
+    except Exception as e:
+        print(traceback.format_exc())
+        raise e
+    return total_inserted
+
+
+def update_operation_entities(mongo_source, collection_name, id_creation_func, filter_criteria, correction_map, batch_size):
+    collection = mongo_source.mongo_handle[mongo_source.db_name][collection_name]
+    total_updated = 0
+    try:
+        for batch_of_variant in find_documents_in_batch(mongo_source, collection_name, filter_criteria, batch_size):
+            update_statements = []
+            for variant in batch_of_variant:
+                filter_dict = {'_id': variant['_id']}
+                filter_dict.update(filter_criteria)
+                for key in correction_map:
+                    if callable(correction_map[key]):
+                        variant[key] = correction_map[key](variant[key])
+                    elif 'inactiveObjects' in key:
+                        inactive_objects = variant['inactiveObjects']
+                        prop = key.split('.')[-1]
+                        for inactive in inactive_objects:
+                            inactive[prop] = correction_map[key]
+                    else:
+                        variant[key] = correction_map[key]
+                for inactive in variant['inactiveObjects']:
+                    inactive['hashedMessage'] = id_creation_func(inactive)
+                variant.pop('_id')
+                update_statements.append(pymongo.ReplaceOne(filter_dict, variant))
+            result_update = collection.with_options(write_concern=WriteConcern(w="majority", wtimeout=1200000)) \
+                .bulk_write(requests=update_statements, ordered=False)
+            total_updated += result_update.modified_count
+        logger.info(f'{total_updated} documents updated in {collection_name}')
+    except Exception as e:
+        print(traceback.format_exc())
+        raise e
+    return total_updated
+
+
+def replace_variant_entities(mongo_source, batch_size):
+    # submitted variants on MT chromosome
+    replace_document_with_correct_information(
+        mongo_source, "submittedVariantEntity", get_submitted_SHA1,
+        {'seq': 'GCA_015227675.1', 'contig': 'CM026996.1'},
+        {'seq': 'GCA_015227675.2', 'contig': 'AY172581.1'},
```

Comment:
Given that we are repeating these values many times in the script, suggest extracting them to a variable...